### PR TITLE
New Package: NTPoly

### PR DESCRIPTION
**Pull request by william-dawson**
I would like to contribute a new package for SPACK called NTPoly.

NTPoly is a massively parallel library for computing the functions of sparse, hermitian matrices based on polynomial expansions. For sufficiently sparse matrices, most of the matrix functions in NTPoly can be computed in linear time.

I would like to thank the SPACK developers for giving a tutorial a few weeks ago at RIKEN, and I hope that this PR lines up with what they taught us!

### CODE REVIEW

File: var/spack/repos/builtin/packages/ntpoly/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    NTPoly can be computed in linear time.
+    """
+
+    homepage = "https://william-dawson.github.io/NTPoly/"
+    url      = "https://github.com/william-dawson/NTPoly/archive/ntpoly-v2.3.1-alpha.tar.gz"
+
+    version('2.3.1', sha256='af8c7690321607fbdee9671b9cb3acbed945148014e0541435858cf82bfd887e')
+
+    depends_on('cmake', type='build')
+    depends_on('blas', type='build')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/ntpoly/package.py]: I'm guessing this is actually a link dependency?

File: var/spack/repos/builtin/packages/ntpoly/package.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    depends_on('mpi@3', type='build')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/ntpoly/package.py]: this too

**Comment by william-dawson** [comment]
@tgamblin thanks for the feedback. I've made some modifications according to your suggestions. Let me know if there is anything else I can do better.

(I'm not sure what is going on with the code coverage analysis)

**Comment by scheibelp** [comment]
Thanks!

> I'm not sure what is going on with the code coverage analysis

We ignore the code coverage stats for package PRs, so you don't have to worry about that

**Pull request merged**